### ISSUE TITLE
Configurable ReturnQuotaExceededResponse content

### ISSUE THREAD

**Issue by lestevaos** [stefanprodan/AspNetCoreRateLimit#14]
Hello Stefan!

First of all, thank your very much for this middleware. It's extremely useful. Big deal. True. /trump

I was thinking if it would be a good idea allow to change the response from the `ReturnQuotaExceededResponse` method without having to override it.

In my current API project, I want to reply always with a very specific JSON body format like:
```
{
    "id": "too_many_requests",
    "message": "Whoa! Calm down, cowboy!",
    "url": "https://myapi.com/docs/errors#too_many_requests",
    "details": "Quota exceeded. Maximum allowed: {rule.Limit} per {rule.Period}. Please try again in {retryAfter} seconds."
}
```

I'm aware that the string interpolation is not going to work if we get the output format from the configuration via options, but maybe the good old positional string format placeholders can do the trick.

So in the appsettings.json file:
```
"ClientRateLimiting": {
    ...
    "QuotaExceededResponse": {
        "ContentType": "application/json",
        "Content": "{{ \"id\": \"too_many_requests\", \"message\": \"Whoa! Calm down, cowboy!\", \"url\": \"https://myapi.com/docs/errors#too_many_requests\", \"details\": \"Quota exceeded. Maximum allowed: {0} per {1}. Please try again in {2} seconds.\" }}"
    }
}
```
And in the ClientRateLimitMiddleware.cs we will have:
```
public virtual Task ReturnQuotaExceededResponse(HttpContext httpContext, RateLimitRule rule, string retryAfter)
{
    var response = new QuotaExceededResponse
    {
        Content = $"API calls quota exceeded! maximum admitted {rule.Limit} per {rule.Period}."
        ContentType = "text/plain"
    };

    if (_options.QuotaExceededResponse != null)
    {
        // {0} rule.Limit
        // {1} rule.Period
        // {2} retryAfter
        // ... maybe all the rule properties?
        response.Content = string.Format(_options.QuotaExceededResponse.Content, rule.Limit, rule.Period, retryAfter);
        response.ContentType = _options.QuotaExceededResponse.ContentType;
    }

    if (!_options.DisableRateLimitHeaders)
    {
        httpContext.Response.Headers["Retry-After"] = retryAfter;
    }

    httpContext.Response.StatusCode = _options.HttpStatusCode;

    return httpContext.Response.WriteAsync(response);
}
```

Sorry if there's a typo or if the code does not compile. I just wrote it on the fly here.

Thank you for your time.

**Comment by boformer** [comment]
That would be a nice addition!

**Comment by cristipufu** [comment]
Great suggestion! Fixed by https://github.com/stefanprodan/AspNetCoreRateLimit/commit/b467dde6d0e652f4b55e026e4ae7fc18ff334349.

You can now configure the quota exceeded response:
```json
"QuotaExceededResponse": {
  "Content": "{{ \"message\": \"Whoa! Calm down, cowboy!\", \"details\": \"Quota exceeded. Maximum allowed: {0} per {1}. Please try again in {2} second(s).\" }}",
  "ContentType": "application/json",
  "StatusCode": 429
}
```

**Comment by reyhan68** [comment]
> Great suggestion! Fixed by [b467dde](https://github.com/stefanprodan/AspNetCoreRateLimit/commit/b467dde6d0e652f4b55e026e4ae7fc18ff334349).
> 
> You can now configure the quota exceeded response:
> 
> ```json
> "QuotaExceededResponse": {
>   "Content": "{{ \"message\": \"Whoa! Calm down, cowboy!\", \"details\": \"Quota exceeded. Maximum allowed: {0} per {1}. Please try again in {2} second(s).\" }}",
>   "ContentType": "application/json",
>   "StatusCode": 429
> }
> ```

I tested your answer, it didn't work
 "QuotaExceededResponse": {
      "Content": "{{ \"message\": \"Whoa! Calm down, cowboy!\", \"details\": \"My Custom Error: {0} per {1}. Please try again in {2} second(s).\" }}",
      "ContentType": "application/json"
    },
![image](https://user-images.githubusercontent.com/50240584/217762906-3ee0a160-7cde-411f-82f5-326a1c098991.png)
